core/trino-main/src/main/java/io/trino/operator/aggregation/OrderedAccumulatorFactory.java

```diff
@@ -211,7 +211,16 @@ public void addInput(GroupByIdBlock groupIdsBlock, Page page, Optional<Block> ma
             if (mask.isPresent()) {
                 page = filter(page, mask.orElseThrow());
             }
-            pagesIndex.addPage(page);
+            if (page.getPositionCount() == 0) {
```

Comment:
the same seems to be required for `DistinctAccumulator`

Reply:
Good catch.  I looked at DistinctAccumulator also, and missed the place where it skips input

Reply:
The existing distinct code is correct, but I added a test to verify that we don't introduce this bug in that code (also, I verified the test catches the bug if we change the code).  I also added a comment to the distinct code to remind people about the issue